### ISSUE TITLE
Unable to uninstall .NET Core 3.1.20 Hosting Bundle.

### ISSUE THREAD

**Issue by DM2489** [dotnet/runtime#62853]
### Description

Hi,

I'm unable to uninstall .NET Core 3.1.20 on one of my web servers. The uninstaller crashes part way through the process.

I have also tried the .NET Core uninstall tool, it has the same issue.

Event Viewer gives the following:

Faulting application name: WindowsServerHostingBundle.exe, version: 3.1.20.21472, time stamp: 0x5ec2f717
Faulting module name: wixstdba.dll, version: 3.14.4118.0, time stamp: 0x5ec2f758
Exception code: 0xc0000005
Fault offset: 0x00003c80
Faulting process id: 0x1f7c
Faulting application start time: 0x01d7f1e47c4340c3
Faulting application path: C:\ProgramData\Package Cache\{1387af25-2c3c-4213-bdd8-443298750e72}\WindowsServerHostingBundle.exe
Faulting module path: C:\WINDOWS\Temp\{1E7D1A85-E23B-49AB-B8D4-B8BB2E41A283}\.ba\wixstdba.dll
Report Id: fd96602a-cca1-42fe-9b92-6150f7da9d78
Faulting package full name: 
Faulting package-relative application ID: 

### Reproduction Steps

Control Panel -> right click .NET Core 3.1.20 hosting bundle -> uninstall.

All other servers uninstalled fine. 

### Expected behavior

To be able to uninstall the framework.

### Actual behavior

Crashes and burns, unable to uninstall.

**Comment by vitek-karas** [comment]
@ericstj I think the `WindowsServerHostingBundle` is the ASP.NET native module, right? In that case this issue would belong to the aspnet repo?

**Comment by rseanhall** [comment]
There should be a log file in %TEMP%, called something like dd_dd_DotNetCoreWinSvrHosting__<timestamp>.log. Can you provide that?

**Comment by DM2489** [comment]
[dd_dd_DotNetCoreWinSvrHosting__20211215180432.log](https://github.com/dotnet/runtime/files/7722852/dd_dd_DotNetCoreWinSvrHosting__20211215180432.log)

Apologies, we tried many times to uninstall, then repair the install, so log folder was quite polluted. I believe this one is a failed run.

However, it is a moot point now. I just re-tried the uninstaller and it worked without fail. The only change that was made was installing the .NET Core 6.0.1 hosting bundle.

Thanks for the assistance anyway.

**Comment by rseanhall** [comment]
wixstdba.dll shouldn't crash. If it created a dump file, can you upload that with the log to a new issue at https://github.com/wixtoolset/issues?

@ericstj This is probably https://github.com/wixtoolset/wix3/pull/531, you guys might want to update to the latest WiX v3.14 for 3.1.

**Comment by vitek-karas** [comment]
Thanks a lot @rseanhall for helping with this!

**Comment by ericstj** [comment]
Thank you @rseanhall.  We did just update Wix in our 3.1/5.0/6.0 branches in order to get the ARM64 changes.  The first 3.1 release with this would have been 3.1.21: https://dotnet.microsoft.com/en-us/download/dotnet/3.1

@DM2489 as a workaround can you try upgrading to a newer version then uninstalling?  That might confirm the fix while also unblocking you.

cc @NikolaMilosavljevic @joeloff

**Comment by DM2489** [comment]
@ericstj Yes, when we installed .NET Core 6.0.1 we were then able to uninstall 3.1.20. Hope that answers your question.

**Comment by ericstj** [comment]
That's interesting, I wouldn't expect 6.0 to help here since it is side-by-side, my suggestion was to use a newer 3.1 -- the theory being that it would replace the uninstall entry for the old bundle without actually running the bootstrapper with the bug.  Then you could uninstall the new 3.1 (3.1.21) since it would be using the new bootstrapper with @rseanhall's fix.

Glad to hear you're not blocked.  I'm going to close this issue since I agree with @rseanhall's assessment that it's likely the linked bug, which should be fixed in the latest 3.1 builds.  For anyone hitting this where the above workaround doesn't fix it let us know and we can reactivate.

**Comment by DM2489** [comment]
Indeed, when I was looking through the log folder I decided to re-run the uninstaller to generate a fresh set of log files, that way I could be confident I had the correct one to upload. But it uninstalled fine. The only change to the server since submitting the issue was installing the .NET Core 6.0.1 hosting bundle, so I assume that must have had something to do with it.

Anyway, thanks for the assistance, glad to be on 3.1.22 and 6.0.1 now.
